pxr/usd/sdf/textFileFormat.ll

```diff
@@ -134,24 +164,42 @@ PXR_NAMESPACE_USING_DIRECTIVE
 "variants"            { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARIANTS; }
 "varying"             { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARYING; }
 
-    /* unquoted C/Python identifier */
-[[:alpha:]_][[:alnum:]_]* {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_IDENTIFIER;
-    }
-
-    /* unquoted C++ namespaced identifier -- see bug 10775 */
-[[:alpha:]_][[:alnum:]_]*(::[[:alpha:]_][[:alnum:]_]*)+ {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_CXX_NAMESPACED_IDENTIFIER;
-    }
-
-    /* unquoted namespaced identifier.  matches any number of colon
-     * delimited C/Python identifiers */
-[[:alpha:]_][[:alnum:]_]*(:[[:alpha:]_][[:alnum:]_]*)+ {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_NAMESPACED_IDENTIFIER;
-    }
+ /* In a Unicode enabled scheme, 'identifiers' are generally
+  * categorized as something that begins with something in the
+  * XID_Start category followed by zero or more things in the
+  * XID_Continue category.  Since the number of characters in
+  * these classes are large, we can't explicitly validate them
+  * here easily, so the lex rule is pretty permissive with some
+  * further validation done in code prior to calling what was
+  * read an 'identifier'.  Note this rule will also match
+  * standard ASCII strings because the UTF-8 encoded byte 
+  * representation is the same for these characters.
+  * However, unlike the path lexer, we can guarantee that 
+  * prim names aren't something special to be called out here
+  * so we can be a little more specific about the kinds of strings
+  * we match, particularly to not collide with the pure digit match rule
+  * below
+  */
+{UTF8NODIGU}{UTF8U}* {
+    (*yylval_param) = std::string(yytext, yyleng);
+    return TOK_IDENTIFIER;
+}
+
+ /* unquoted C++ namespaced identifiers are identifiers separated
+  * by the '::' character -- see bug 10775.
+  */
+{UTF8NODIGU}{UTF8U}*(::{UTF8NODIGU}{UTF8U}*)+ {
+    (*yylval_param) = std::string(yytext, yyleng);
+    return TOK_CXX_NAMESPACED_IDENTIFIER;
```

Comment:
@gitamohr In the spirit of your note about not adding UTF-8 support to mappers should, we leave CXX_NAMESPACED_IDENTIFIERs alone too?

Reply:
I'm okay with that, I think.